src/data/metadata/MetadataD2ApiRepository.ts

```diff
@@ -86,11 +86,10 @@ export class MetadataD2ApiRepository implements MetadataRepository {
                 metadataTransformations
             );
 
-            if (dataStoreIds.length > 0) {
-                metadataPackage.dataStores = await d2ApiDataStore.getDataStore({ namespaces: ids });
-            }
+            const dataStoresMetadata = await this.getDataStoresMetadata(ids);
+            const responseWithDataStores = { ...metadataPackage, ...dataStoresMetadata } as T;
```

Comment:
[note] just for the record, this `as T` is type-unsafe, but the whole application uses this method, so unfortunately nothing to do here...